lib/node.js

```diff
@@ -415,7 +415,54 @@ declare module "events" {
 }
 
 declare module "fs" {
-  // TODO
+  declare class Stats {
+    // TODO: fill this
+  }
+
+  declare function rename(oldPath: string, newPath: string, callback: (err: ?Error) => void): void;
+  declare function renameSync(oldPath: string, newPath: string): void;
+  declare function stat(path: string, callback: (err: ?Error, stats: Stats) => void): void;
+  declare function statSync(path: string): Stats;
+  declare function unlink(path: string, callback: (err: ?Error) => void): void;
+  declare function unlinkSync(path: string): void;
+  declare var write: (fd: number, buffer: Buffer, offset: number, length: number, position?: mixed, callback: (err: ?Error, write: number, str: string) => void) => void
```

Comment:
Optional params currently only make sense as the last params of a function. I know what you're trying to say, but I don't think it's going to work like this

```JavaScript
function foo(x: number, y?: number, z?: number) {}
foo(1); // This is fine

function bar(x?: number, y: number, z: number) {}
foo(2,3); // This is an error
```

You could in theory do

```JavaScript
(fd: number, buffer: Buffer, offset: number, length: number, position?: mixed, callback: (err: ?Error, write: number, str: string) => void) => void |
(fd: number, buffer: Buffer, offset: number, length: number, callback: (err: ?Error, write: number, str: string) => void) => void
```

To be completely honest, I think you're supposed to be able to do

```JavaScript
declare function write(fd: number, buffer: Buffer, offset: number, length: number, position: mixed, callback: (err: ?Error, write: number, str: string) => void): void;
declare function write(fd: number, buffer: Buffer, offset: number, length: number, callback: (err: ?Error, write: number, str: string) => void): void;
...etc
```